align/compiler/write_constraint.py

```diff
@@ -267,14 +272,16 @@ def compare_nodes(G,match_pair,traversed,node1,node2, ports_weight):
 
     """
     logger.debug("comparing %s,%s, traversed %s",node1,node2,traversed)
-    nbrs1 = set(G.neighbors(node1)) - set(traversed)
-    nbrs2 = set(G.neighbors(node2)) - set(traversed)
+    nbrs1 = sorted(set(G.neighbors(node1)) - set(traversed))
```

Comment:
Missing key to sorted? Or does the default sort comparison work well enough here?

Reply:
This sort ensures between runs the results don't change. It's not required for the algo.